imgui/src/drag_drop.rs

```diff
@@ -0,0 +1,475 @@
+        /// Request hiding the igBeginDragDropSource tooltip from the igBeginDragDropTarget site.
+        const ACCEPT_NO_PREVIEW_TOOLTIP = sys::ImGuiDragDropFlags_AcceptNoPreviewTooltip;
+        /// For peeking ahead and inspecting the payload before delivery. This is just a convenience
+        /// flag for the intersection of `ACCEPT_BEFORE_DELIVERY` and `ACCEPT_NO_DRAW_DEFAULT_RECT`
+        const ACCEPT_PEEK_ONLY = sys::ImGuiDragDropFlags_AcceptPeekOnly;
+    }
+);
+
+/// Creates a source for drag drop data out of the last ID created.
+///
+/// ```no_run
+/// # use imgui::*;
+/// fn show_ui(ui: &Ui<'_>) {
+///     ui.button(im_str!("Hello, I am a drag source!"), [0.0, 0.0]);
+///     
+///     // Creates an empty DragSource with no tooltip
+///     DragDropSource::new(im_str!("BUTTON_DRAG")).begin(ui);
+/// }
+/// ```
+///
+/// Notice especially the `"BUTTON_DRAG"` name -- this is the identifier of this
+/// DragDropSource; [DragDropTarget]'s will specify an identifier to *receive*, and these
+/// names must match up. A single item should only have one [DragDropSource], though
+/// a target may have multiple different targets.
+///
+/// DropDropSources don't do anything until you use one of the three `begin_` methods
+/// on this struct. Each of these methods describes how you handle the Payload which ImGui
+/// will manage, and then give to a [DragDropTarget], which will received the payload. The
+/// simplest and safest Payload is the empty payload, created with [begin](Self::begin).
+#[derive(Debug)]
+pub struct DragDropSource<'a> {
+    name: &'a ImStr,
+    flags: DragDropFlags,
+    cond: Condition,
+}
+
+impl<'a> DragDropSource<'a> {
+    /// Creates a new [DragDropSource] with no flags and the `Condition::Always` with the given name.
+    /// ImGui refers to this `name` field as a `type`, but really it's just an identifier to match up
+    /// Source/Target for DragDrop.
+    pub fn new(name: &'a ImStr) -> Self {
+        Self {
+            name,
+            flags: DragDropFlags::empty(),
+            cond: Condition::Always,
+        }
+    }
+
+    /// Sets the flags on the [DragDropSource]. Only the flags `SOURCE_NO_PREVIEW_TOOLTIP`,
+    /// `SOURCE_NO_DISABLE_HOVER`, `SOURCE_NO_HOLD_TO_OPEN_OTHERS`, `SOURCE_ALLOW_NULL_ID`,
+    /// `SOURCE_EXTERN`, `SOURCE_AUTO_EXPIRE_PAYLOAD` make semantic sense, but any other flags will
+    /// be accepted without panic.
+    pub fn flags(mut self, flags: DragDropFlags) -> Self {
+        self.flags = flags;
+        self
+    }
+
+    /// Sets the condition on the [DragDropSource]. Defaults to [Always](Condition::Always).
+    pub fn condition(mut self, cond: Condition) -> Self {
+        self.cond = cond;
+        self
+    }
+
+    /// Creates the source of a drag and returns a handle on the tooltip.
+    /// This handle can be immediately dropped without binding it, in which case a default empty
```

Comment:
Kind of weird to have this be the one thing that uses permissive dropping behavior already, but IDRC.